File: app/src/main/java/com/knopkapp/activities/RegistrationActivity.java
package com.knopkapp.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.os.Bundle;
import android.widget.ProgressBar;

import com.knopkapp.R;
import com.knopkapp.Registrationandverification1Fragment;
import com.knopkapp.Registrationandverification2Fragment;
import com.knopkapp.Registrationandverification3Fragment;
import com.knopkapp.Registrationandverification4Fragment;

public class RegistrationActivity extends AppCompatActivity {

    private ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_registration);

        progressBar = findViewById(R.id.progressBar);

        if (savedInstanceState == null) {
            setFragment(new Registrationandverification1Fragment());
        }
    }

    public void updateProgressBar(Fragment fragment) {
        if (fragment instanceof Registrationandverification1Fragment) {
            progressBar.setProgress(20);
        } else if (fragment instanceof WriteSMSFragment) {
            progressBar.setProgress(40);
        } else if (fragment instanceof Registrationandverification2Fragment) {
            progressBar.setProgress(60);
        } else if (fragment instanceof Registrationandverification3Fragment) {
            progressBar.setProgress(80);
        } else if (fragment instanceof Registrationandverification4Fragment) {
            progressBar.setProgress(100);
        }
    }

    public void setFragment(Fragment fragment) {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.fragmentContainer, fragment)
                .commitNow(); // Используйте commitNow для немедленного выполнения транзакции

        updateProgressBar(fragment); // Обновите прогресс после завершения транзакции
    }

}

File: app/src/main/java/com/knopkapp/activities/RegistrationActivity.kt
package com.knopkapp.activities

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

import androidx.navigation.findNavController
import com.google.firebase.firestore.DocumentReference
import com.google.firebase.firestore.FirebaseFirestore
import com.knopkapp.R
import com.knopkapp.databinding.ActivityRegistrationBinding
import com.knopkapp.db.SessionManager
import com.knopkapp.models.UniversalDate
import com.knopkapp.waiter.WaiterMainScreenActivity
import com.knopkapp.waiter.WaiterTablesActivity

class RegistrationActivity : AppCompatActivity() {

    private lateinit var binding: ActivityRegistrationBinding
    lateinit var firebaseFireStore: FirebaseFirestore

    private lateinit var sessionManager: SessionManager
    private lateinit var directorDocument: DocumentReference
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityRegistrationBinding.inflate(layoutInflater)
        setContentView(binding.root)
        sessionManager = SessionManager(this)
        firebaseFireStore = FirebaseFirestore.getInstance()
    }

    override fun onResume() {
        super.onResume()

        if (sessionManager.fio != "") {
            sessionManager.isRegistered = true
            when (sessionManager.status) {
                "Owner" -> {
                    findNavController(R.id.fragmentContaner).navigate(R.id.ownerMainMenuFragment)
                }

                "Director" -> {
                    findNavController(R.id.fragmentContaner).navigate(R.id.directorMainScreenFragment)
                }

                "Administrator" -> {
                    findNavController(R.id.fragmentContaner).navigate(R.id.adminMainMenuFragment)
                }

                "Waiter" -> {
                    val intent = Intent(this, WaiterMainScreenActivity::class.java)
                    startActivity(intent)
                    finish()
                }

            }
        }

    }
}
